executor/newdeploy/newdeploy.go

```diff
@@ -214,6 +246,10 @@ func (deploy *NewDeploy) createOrGetHpa(hpaName string, execStrategy *fission.Ex
 		return existingHpa, err
 	}
 
+	if depl == nil {
+		return nil, errors.New("Failed to create HPA, found empty deployment")
```

Comment:
Minor: try to stick with idiomatic go: error messages should not be capitalized.